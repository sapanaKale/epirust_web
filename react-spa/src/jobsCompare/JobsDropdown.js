/*
 * EpiRust
 * Copyright (c) 2020  ThoughtWorks, Inc.
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 *
 */

import React from "react";
import PropTypes from 'prop-types'

export default function JobsDropdown({jobs, label}) {
  const id = label.split(' ').join('').toLowerCase();
  return <>
    <label htmlFor={id}>{label}</label>
    <select name={id} className="form-control" id={id} required>
      <option disabled selected value=""> -- select a job -- </option>
      {jobs.map(j => <option key={j._id}>{j._id}</option>)}
    </select>
  </>
}

JobsDropdown.propTypes = {
  jobs: PropTypes.array.isRequired,
  label: PropTypes.string.isRequired,
}