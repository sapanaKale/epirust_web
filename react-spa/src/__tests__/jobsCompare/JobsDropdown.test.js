/*
 * EpiRust
 * Copyright (c) 2020  ThoughtWorks, Inc.
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 *
 */

import React from "react";
import {fireEvent, render} from '@testing-library/react'
import JobsDropdown from "../../jobsCompare/JobsDropdown";

describe('Jobs DropDown', () => {
  it('should render dropdown with job id list', function () {
    const jobs = [{_id: "id1"}, {_id: "id2"}]
    const {container} = render(<JobsDropdown jobs={jobs} label="job 1"/>);

    expect(container).toMatchSnapshot()
  });

  it('should not break when dropdown is updated', function () {
    const jobs = [{_id: "id1"}, {_id: "id2"}]
    const {container} = render(<JobsDropdown jobs={jobs} label="job 1"/>);

    expect(() => fireEvent.change(container.querySelector('.form-control'), {target: {value: 'id1'}}))
      .not.toThrow(Error)
  });
});

