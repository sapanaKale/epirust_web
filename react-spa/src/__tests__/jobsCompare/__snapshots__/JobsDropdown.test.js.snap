// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Jobs DropDown should render dropdown with job id list 1`] = `
<div>
  <label
    for="job1"
  >
    job 1
  </label>
  <select
    class="form-control"
    id="job1"
    name="job1"
    required=""
  >
    <option
      disabled=""
      value=""
    >
       -- select a job -- 
    </option>
    <option>
      id1
    </option>
    <option>
      id2
    </option>
  </select>
</div>
`;
